src/lib.rs

```diff
@@ -115,6 +115,75 @@ fn main() -> std::io::Result<()> {
 }
 ```
 
+## Custom `Registry`
+
+Some apps might have more than one `actix_web::HttpServer`.
+If that's the case, you might want to use your own registry:
+
+```rust
+use actix_web::{web, App, HttpResponse, HttpServer};
+use actix_web_prom::PrometheusMetrics;
+use prometheus::Registry;
+use std::thread;
+
+fn test1() -> HttpResponse {
+    HttpResponse::Ok().body("Everyone can see it!")
+}
+fn test2() -> HttpResponse {
+    HttpResponse::Ok().body("This can be hidden behind a firewall")
+}
+
+fn main() -> std::io::Result<()> {
+    let shared_registry = Registry::new();
+
+    let private_prom = PrometheusMetrics::new_with_registry(
+                                        shared_registry.clone(),
+                                        "private_api",
+                                        Some("/metrics")
+                                    )
+                                    // It is safe to unwrap when __no other app has the same namespace__
+                                    .unwrap();
+    let public_prom = PrometheusMetrics::new_with_registry(
+                                        shared_registry.clone(),
+                                        "public_api",
+                                        // Metrics should not be availible from the outside
+                                        None
+                                    )
+                                    .unwrap();
+
+# if false {
+    let public_handle = thread::spawn(move || {
+        HttpServer::new(move || {
+            App::new()
+                .wrap(private_prom.clone())
+                .service(web::resource("/test").to(test1))
```

Comment:
I think the way this is exposed (without a bit more docs) can be confusing. You have a public handle that has a "private prom" middleware, for which said private_prom is named "private_api" with `/metrics` but the function responds "Everyone can see it".

That being the intent, it's hard to follow.

Reply:
My bad, should've named these functions not `test1` and `test2`
Yes, server with `public_prom` should return "Everyone can see it"